src/Microsoft.Identity.Web/WebAppExtensions/MicrosoftIdentityWebAppAuthenticationBuilderWithConfiguration.cs

```diff
@@ -57,8 +59,6 @@ public class MicrosoftIdentityWebAppAuthenticationBuilderWithConfiguration : Mic
                         options.ClientSecret = AppServicesAuthenticationInformation.ClientSecret;
                         options.Instance = AppServicesAuthenticationInformation.Issuer;
                     }
-
-                    Services.AddHttpClient();
```

Comment:
At this point, in the callback, the service collection is already readonly (the Dependency injection has happened). This is now enforced in .NET 8 rc1.

Services.AddHttpClient() needs to move outside of the callback